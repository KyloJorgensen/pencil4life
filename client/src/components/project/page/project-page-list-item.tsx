'use strict';

import * as React from 'react';
import { Link } from 'react-router-dom';
import * as objectPath from 'object-path';
import ImageTag from '../../image/image-tag';
import { projectPageConsumer, IProjectPageContext } from './project-page-provider';

export interface ProjectPageListItemProps {
	projectPage: IProjectPageContext;
	_projectPageId: string;
}

class ProjectPageListItem extends React.Component<ProjectPageListItemProps> {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
    	this.props.projectPage.getProjectPage(this.props.projectPage._projectId, this.props._projectPageId);
    }

	render() {
		const { _projectId } = this.props.projectPage;
		const { _projectPageId, title, _imageId } = this.props.projectPage.projectPage;
		const imageIdElem = _imageId ? (<ImageTag _imageId={_imageId} />) : '';
		
		return (
			<div className="project-page-list-item-wrapper" >				
				<Link className="project-link" to={'/project/item/'+_projectId+'/'+_projectPageId} >
					<div className="image-wrapper">
						{imageIdElem}
					</div>
				</Link>
			</div>
		);		
	}
};

export default projectPageConsumer(ProjectPageListItem);
export {ProjectPageListItem};