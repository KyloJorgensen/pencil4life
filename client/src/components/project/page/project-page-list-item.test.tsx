'use strict';

import * as React from 'react';
import { shallow } from 'enzyme';
import * as RichTextEditor from 'react-rte'
import { ProjectPageListItem, ProjectPageListItemProps } from './project-page-list-item';

const _defaultProps: ProjectPageListItemProps = {
    projectPage: {    
        getProjectPages: (params) => {},
        getProjectPage: (params) => {},
        addProjectPage: (params) => {},
        updateProjectPage: (params, callback) => {},
        projectPages: [],
        projectPageList: {},
        page: 0,
        totalPages: 0,
        limit: 0,
        total: 0,
        pageTotal: 0,
        projectPage: {
            _projectPageId: '',
            title: '',
            details: RichTextEditor.createEmptyValue().toString('html'),
            discontinued: false,
            _imageId: '',
            page: 0,
        }
    },
    _projectPageId: '',
};

describe('Render Project Page List Item', () => {
	it('matches the snapshot', () => {
		let wrapper = shallow(<ProjectPageListItem {..._defaultProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});