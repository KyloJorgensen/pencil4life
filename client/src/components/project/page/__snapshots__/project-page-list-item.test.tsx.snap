// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Project Page List Item matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProjectPageListItem
    _projectPageId=""
    projectPage={
      Object {
        "addProjectPage": [Function],
        "getProjectPage": [Function],
        "getProjectPages": [Function],
        "limit": 0,
        "page": 0,
        "pageTotal": 0,
        "projectPage": Object {
          "_imageId": "",
          "_projectPageId": "",
          "details": "<p><br></p>",
          "discontinued": false,
          "page": 0,
          "title": "",
        },
        "projectPageList": Object {},
        "projectPages": Array [],
        "total": 0,
        "totalPages": 0,
        "updateProjectPage": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="project-link"
        replace={false}
        to="/project/item/undefined/"
      >
        <div
          className="image-wrapper"
        >
          
        </div>
      </Link>,
      "className": "project-page-list-item-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="image-wrapper"
        >
          
        </div>,
        "className": "project-link",
        "replace": false,
        "to": "/project/item/undefined/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "image-wrapper",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="project-link"
          replace={false}
          to="/project/item/undefined/"
        >
          <div
            className="image-wrapper"
          >
            
          </div>
        </Link>,
        "className": "project-page-list-item-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="image-wrapper"
          >
            
          </div>,
          "className": "project-link",
          "replace": false,
          "to": "/project/item/undefined/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "image-wrapper",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
